src/utils/fsUtils.ts

```diff
@@ -1,82 +1,90 @@
-import fs from 'fs'
+import fs from 'fs-extra'
 import path from 'path'
 
-const joinPaths = (filepaths: string[]): string =>
-  path.join(...filepaths.filter(Boolean))
+const joinPaths = (filepaths: string | string[]): string =>
+  Array.isArray(filepaths) ? path.join(...filepaths) : filepaths
 
 /**
  * tests whether or not the given path exists by checking with the file system.
- * @param filepaths Joins all arguments as file paths
+ * @param filepath path or array with paths to join with `path.join`
  * ```js
- * pathExists("dir", "file.ext")
+ * pathExists(["dir", "file.ext"])
  * ```
  */
-export const pathExists = (...filepaths: string[]): boolean =>
-  fs.existsSync(joinPaths(filepaths))
+export const pathExists = (filepath: string | string[]): boolean =>
+  fs.existsSync(joinPaths(filepath))
 
 /**
  * Read file contents as a string (UTF-8)
- * @param filepaths Joins all arguments as file paths
+ * @param filepath path or array with paths to join with `path.join`
  * ```js
- * readFile("dir", "file.ext")
+ * readFile(["dir", "file.ext"])
  * ```
  */
-export const readFile = (...filepaths: string[]): string =>
-  fs.readFileSync(joinPaths(filepaths), 'utf8')
+export const readFile = (filepath: string | string[]): string =>
+  fs.readFileSync(joinPaths(filepath), 'utf8')
 
 /**
  * Read file contents as string or if the path doesn't exists returns undefined
- * @param filepaths Joins all arguments as file paths
+ * @param filepath path or array with paths to join with `path.join`
  * ```js
- * readFileIfExists("dir", "file.ext")
+ * readFileIfExists(["dir", "file.ext"])
  * ```
  */
-export const readFileIfExists = (...filepaths: string[]): string | undefined =>
-  pathExists(...filepaths) ? readFile(...filepaths) : undefined
+export const readFileIfExists = (
+  filepath: string | string[]
+): string | undefined => (pathExists(filepath) ? readFile(filepath) : undefined)
```

Comment:
This for me gets a bit hard to read, with all the typings.

Maybe I'm just not used to it, but what do you think about splitting it up so it has a brace and `return` statement?